routes/profile.js

```diff
@@ -1,136 +1,118 @@
-      rowWithConfig.numGeoids = match.geoids.length;
-
-      // this is a special metadata object that lets the client
-      // know which estimates were bottom coded with which direction.
-      // The `delegateAggregator` special function `interpolate`
-      // specifically assigns properties to this object.
-      rowWithConfig.codingThresholds = {};
-
-      // if the row is "special" and the number of geoids in the
-      // selection are greater than 1
-      // then, delete the unneeded special calculations data
-
-      if (rowWithConfig.special && (match.geoids.length > 1)) {
-        const currentYear = get(fullDataset, dataset);
-        const newRowObject = delegateAggregator(rowWithConfig, rowWithConfig.rowConfig, currentYear);
-        rowWithConfig = merge(newRowObject, rowWithConfig);
-      }
-
-      return rowWithConfig;
-    });
-};
-
-const appendIsReliable = data => (data.map((row) => {
-  const appendedRow = row;
-  appendedRow.is_reliable = false;
-  appendedRow.comparison_is_reliable = false;
-
-  const {
-    cv,
-    comparison_cv,
-    codingThresholds,
-    change_m,
-    change_sum,
-    change_percent_m,
-    change_percent,
-    difference_sum,
-    difference_m,
-  } = appendedRow;
-
-  // set reliability to true if cv is less than 20
-  if (cv !== null && cv < 20) appendedRow.is_reliable = true;
-  if (comparison_cv !== null && comparison_cv < 20) appendedRow.comparison_is_reliable = true;
-
-  // set reliability to false if the value is top or bottom-coded
-  if (codingThresholds.sum) appendedRow.is_reliable = false;
-  if (codingThresholds.comparison_sum) appendedRow.comparison_is_reliable = false;
-
-  // !!!!!! These calculations were changed from determining statistical SIGNIFICANCE to determining statistical RELIABILITY in PR #39, variable names were NOT updated to reflect this calculation change
-  // !!!!!! These names will be updated in a broader refactor of PFF, Visit issue #57 in labs-factfinder waffle to get more info
-  appendedRow.significant = ((((difference_m) / 1.645) / Math.abs(difference_sum)) * 100) < 20;
-  appendedRow.change_significant = ((((change_m) / 1.645) / Math.abs(change_sum)) * 100) < 20;
-  appendedRow.change_percent_significant = ((((change_percent_m) / 1.645) / Math.abs(change_percent)) * 100) < 20;
-
-  return appendedRow;
-}));
-
-const invalidCompare = (compare) => {
+/*
+ * Queries postgres for current, previous, and compare data for a given
+ * profile type, set of geoids and compare geoid. Joins the data, and adds
+ * 'change' and 'difference' calculation values.
+ * @param {string} profile - The profile type
+ * @param {Array} geoids - The list of geoids for the given selected geography
+ * @param {string} compare - Integer string representing the geoid of the comparison geography
+ * @returns {Object}
+ */
+async function getProfileData(profile, geoids, compare, db) {
+  const isAggregate = geoids.length > 1;
+
+  const queryBuilder = getQueryBuilder(profile);
```

Comment:
See my comment above about the naming here - I'm following after a bit of confusion around when something is really a full `profile` or just the name of a profile, like part of an enum or symbol or something.